tests/parser/CompositionalTests.cpp

```diff
@@ -327,4 +327,280 @@ BOOST_AUTO_TEST_CASE(CompositionalParsingTest) {
     }
 }
 
+Deck createCompositionalDeckZMF()
+{
+    return Parser{}.parseString(R"(
+------------------------------------------------------------------------
+RUNSPEC
+------------------------------------------------------------------------
+TITLE
+   SIMPLE CO2 CASE FOR PARSING TEST
+
+METRIC
+
+TABDIMS
+8* 2 3/
+
+OIL
+GAS
+DIMENS
+4 1 1
+/
+
+COMPS
+3 /
+
+------------------------------------------------------------------------
+GRID
+------------------------------------------------------------------------
+DX
+4*10
+/
+DY
+4*1
+/
+DZ
+4*1
+/
+
+TOPS
+4*0
+/
+
+
+PERMX
+4*100
+/
+
+PERMY
+4*100
+/
+
+PERMZ
+4*100
+/
+
+PORO
+1. 2*0.1  1.
+/
+
+------------------------------------------------------------------------
+PROPS
+------------------------------------------------------------------------
+
+CNAMES
+DECANE
+CO2
+METHANE
+/
+
+ROCK
+68 0 /
+
+EOS
+PR /
+SRK /
+
+BIC
+0
+1 2 /
+1
+2 3 /
+
+ACF
+0.4 0.2 0.01 /
+0.5 0.3 0.03 /
+
+PCRIT
+20. 70. 40. /
+21. 71. 41. /
+
+TCRIT
+600. 300. 190. /
+601. 301. 191. /
+
+MW
+142.  44.  16. /
+142.1 44.1 16.1 /
+
+VCRIT
+0.6  0.1  0.1 /
+0.61 0.11 0.11 /
+
+
+STCOND
+15.0 /
+
+SGOF
+-- Sg    Krg    Kro    Pcgo
+   0.0   0.0    1.0    0.0
+   0.1   0.1    0.9    0.0
+   0.2   0.2    0.8    0.0
+   0.3   0.3    0.7    0.0
+   0.4   0.4    0.6    0.0
+   0.5   0.5    0.5    0.0
+   0.6   0.6    0.4    0.0
+   0.7   0.7    0.3    0.0
+   0.8   0.8    0.2    0.0
+   0.9   0.9    0.1    0.0
+   1.0   1.0    0.0    0.0
+/
+
+
+------------------------------------------------------------------------
+SOLUTION
+------------------------------------------------------------------------
+
+PRESSURE
+1*150 2*75. 1*37.5
+/
+
+SGAS
+4*1.
+/
+
+TEMPI
+4*150
+/
+
+ZMF
+1*0.99 3*0.5
+1*0.009 3*0.3
+1*0.001 3*0.2
+/
+
+END
+)");
+}
+// TODO: later might decide which deck to keep, each one evolve to different directions based on new understanding.
+//  for now, we keep both two.
```

Comment:
If we're going to say something here at all, then we need to be more specific about what "both" means.

Reply:
Yes. I updated the comments.